### ISSUE TITLE
[Tracker] v1 Mainnet Release Requirements

### ISSUE THREAD

**Issue by drwasho** [OpenBazaar/OpenBazaar-Client#965]
These are the following issues, group according to categories, that need to be resolved before we can make officially launch v1 on the mainnet.

---
### Priority Ranking

:star: :star: :star: **High priority** 
- _Primary issues to commit and test asap_
- Mainnet release will not occur without these issues resolved

:star: :star: **Medium priority**
- _Secondary issues to fix and test_
- Mainnet release possible, not but advisable without these issues resolved

:star: **Low priority**
_Peripheral issues; save for last_
- Mainnet release not ideal, but not too bad without these issues resolved

---
### Blockchain ID
- [x] #303 ~~Blockchain ID in Chat~~
  - :star: :star:
- [ ] #715 Click to register handle via Onename
  - :star: :star:
### Notifications
- [x] #336 ~~Libbitcoin server is down~~
  - :star: :star:
- [ ] #804 Various notifications 
  - [ ]  Nudge buyer to release funds on a transaction (x days after order date) 
    - :star: :star:
  - [ ]  Moderator revoked their service from your store 
    - :star: :star:
  - [ ]  A user added you as Moderator 
    - :star:
  - [ ]  Moderator rate changes (send to all vendors using their service) 
    - :star:
- [x] #899 ~~Wire up review notification~~
  - :star: :star: 
- [x] #944 ~~Show network error in status bar~~
  - :star: :star: :star:
- [x] #932 Moderator can't see their message when they close a dispute
  - :star: :star:
- [x] #964 ~~Discussion Tab notifications for messages~~
  - :star: :star: :star:
### Listings
- [x] #351 ~~Worldwide shipping (and make default selection instead of 'United States')~~
  - :star: :star: :star:
- [x] #501 ~~Free domestic shipping and paid international shipping option (remove minimums)~~
  -  :star: :star: :star:
- [x] #942 ~~Rating descriptions need to be added~~
  - :star: :star:
- [x] #928 ~~Fix up filters of transactions in 'Purchases'~~
  - :star: :star: :star: 
- [x] #966 ~~Auto update listings to include a newly added moderator~~
  - :star: :star: :star: 
- [ ] #1133 Stock units for listings
  - :star: :star: :star:
### Transaction Flow
- [ ] #838 Remove 'accept dispute payout' for losing party 
  - :star:  
- [x] #914 ~~Show Moderator fee in selection screen~~
  - :star: :star: :star:
- [x] #1053 ~~Hide order modal after clicking 'save'~~
  - :star: :star: :star:
### Reputation and Ratings
- [x] #840 ~~Reputation functions: link to txid on block explorer~~
  - :star: :star: :star: 
- [x] #860 ~~Enable Buyer ratings if they win a dispute~~
  - :star: :star: :star: 
### Design
- [x] #367 ~~Full screen mode~~
  - :star:
- [x] #456 ~~Vendor/seller terminology consistency~~
  - :star:
- [ ] #592 Wire up filters for following/followers 
  - :star:
- [x] #760 ~~Change the placeholder if no listings found~~
  - :star:
- [x] #881 ~~Avatars not displaying correctly in chat~~
  - :star: :star:

**Comment by drwasho** [comment]
### Week of 7th of March, 2016

@jjeffryes has optimised a lot in the client that wasn't captured in this tracking list. A big one was pagination of followers/following. Another was the ability to disable fancy graphics in the client. Both led to a significant performance improvement. Kudos to Josh.

The following issues were completed since the testnet release last week: #456, #881, #928, #964. We'll continue working through this list. Another key optimisation that needs to be done on the server side (see https://github.com/OpenBazaar/OpenBazaar-Server/issues/279) is to enable caching of the images in Electron for our API image calls. This should lead to a significant improvement in the performance of the client connecting to remote nodes.

We also found some consensus on the individual ratings data, which was to include a block explorer link to the associated Bitcoin transaction (txid) and make the inclusion of the rater's GUID `opt-in`.

---

Based on the importance indicators above, these are the remaining issues **must** be resolved prior to any mainnet release (i.e. no mainnet release without them):
1. #944 ~~Show network error in status bar~~
2. #351 ~~Worldwide shipping (and make default selection instead of 'United States')~~
3. #501 ~~Free domestic shipping and paid international shipping option (remove minimums)~~
4. #966 Auto update listings to include a newly added moderator
5. #840 Reputation functions: optional GUID, link to txid on block explorer
6. #860 Enable Buyer ratings if they win a dispute
7. #1053 Hide order modal after clicking 'save'
8. #914 ~~Show Moderator fee in selection screen~~

**Comment by drwasho** [comment]
### Week of 15th of March, 2016

Wow, what a productive week! Lots of issues closed and we're halfway done with the major mainnet issues to be fixed.

**Updated list of priority mainnet requirements:**
 I'm now indicating which of these issues requires corresponding server modifications for @cpacia: 
1. #966 Auto update listings to include a newly added moderator
   - :wrench: Server
2. #840 Reputation functions: optional GUID, link to txid on block explorer
   - :wrench: Server (just the optional GUID part)
3. #860 Enable Buyer ratings if they win a dispute
   - :wrench: Server
4. #1053 ~~Hide order modal after clicking 'save'~~

**Comment by drwasho** [comment]
### Week of 21st of March, 2016

Lots improvements and fixes have been made. We're now in the phase of coming to some internal consensus regarding the issues we cannot launch without. 

**Updated list of priority mainnet requirements:**
1. #966 ~~Auto update listings to include a newly added moderator~~
   - :wrench: Server
2. #840 ~~Reputation functions: link to multisig address on block explorer~~
3. #860 Enable Buyer ratings if they win a dispute
   - :wrench: Server
4. #1133 Stock units for listings
   - Last minute addition based on feedback we've received from the community
   - We enable users to purchase items by quantity, but we currently don't allow users to indicate how much stock of that item they have
   - :wrench: Server

**Comment by drwasho** [comment]
### Week of 28th of March, 2016

Getting much closer to launch. Some key pieces came together on the server side over the weekend, which should clear a path to hopefully knocking over the remaining issues left for mainnet this week and next week.

**Updated list of priority mainnet requirements:**
1. #860 ~~Enable Buyer ratings if they win a dispute~~
2. #1133 Stock units for listings
   - Last minute addition based on feedback we've received from the community
   - We enable users to purchase items by quantity, but we currently don't allow users to indicate how much stock of that item they have
   - :wrench: Server
   - This is a stretch goal, but one that we'd really like to push for.

**Comment by drwasho** [comment]
Ok... we've accomplished all the 'can't launch without' features for OpenBazaar v1. Stock units for listings is still a stretch goal, but it's looking more like something we'll roll out shortly after the mainnet release.
